Dockerfile

```diff
@@ -1,28 +1,56 @@
-FROM python:3.12-slim
+ARG BASE_IMAGE=python:3.12-slim
+FROM $BASE_IMAGE AS base
 
+ARG REQUIREMENTS=requirements-production.txt
+
+# Set environment variables
+ENV FLASK_APP=govuk-frontend-flask.py
 ENV FLASK_RUN_HOST=0.0.0.0
 ENV FLASK_RUN_PORT=8000
+ENV PYTHONUNBUFFERED=1
 
-# Set the working directory in the container
-WORKDIR /usr/src/app
+# Create a non-root user
+RUN adduser --disabled-password app && \
+    cp /usr/share/zoneinfo/Europe/London /etc/localtime
 
-# Change ownership of the working directory to the non-root user
-RUN chown -R 1000:1000 /usr/src/app
+RUN mkdir /home/app/access
+WORKDIR /home/app/access
+
+# Install node
+RUN apt-get update \
+  && apt-get -y install nodejs npm \
+  && apt-get clean
+COPY package-lock.json .
+COPY package.json .
+RUN npm install
+
+COPY requirements/$REQUIREMENTS requirements.txt
+RUN pip install --upgrade pip
+RUN pip install -r requirements.txt
 
-# Copy the dependencies file to the working directory
-COPY requirements.in ./
 
-# Install any needed dependencies
-RUN pip install --no-cache-dir -r requirements.in
+COPY app ./app
+COPY govuk-frontend-flask.py .
 
-# Copy the project code into the working directory
-COPY . .
+RUN npm run build
+
+# Change ownership of the working directory to the non-root user
+RUN chown -R app:app /home/app
+
+# Cleanup container
+RUN rm -rf /var/lib/apt/lists/*
```

Comment:
This should maybe go higher up under the pip dependencies

Reply:
Also why do we need to this?

Reply:
Jav's recommendation to make the container smaller